### PR TITLE
unit tests for login with new UI

### PR DESCRIPTION
**Pull request by alanb2718**
Unit tests for the login page seem OK now .... please check critically, since this should end up being a prototype for our other UI unit tests as well.

### CODE REVIEW

**Review by jbraswell — APPROVED** [review]

File: src/resources/js/test/loginForm.test.tsx [hunk 1]
```diff
@@ -1,27 +1,136 @@
...
+  } else {
+    const msg = '{ "message": "The provided credentials are incorrect." }';
+    const unicodeMsg = Uint8Array.from(Array.from(msg).map((x) => x.charCodeAt(0)));
+    const strm = new ReadableStream({
+      start(controller) {
+        controller.enqueue(unicodeMsg);
+        controller.close();
+      },
+    });
+    const r: Response = new Response(strm, { status: 401, statusText: 'Unauthorized' });
```
Comment (jbraswell) [inline src/resources/js/test/loginForm.test.tsx:64]: I think you can remove the `: Response` type hint here, and just have:

`const r = new Response(strm, { status: 401, statusText: 'Unauthorized' });`

It should be able to infer the type.

File: src/resources/js/test/loginForm.test.tsx [hunk 2]
```diff
@@ -1,27 +1,136 @@
...
+function mockGetToken(): Token | null {
+  return savedAccessToken;
+}
+
+function mockSetToken(token: Token | null): void {
+  savedAccessToken = token;
+}
+
+async function mockGetUser(params: { userId: number }): Promise<User> {
+  if (params.userId == mockUser.id) {
```
Comment (jbraswell) [inline src/resources/js/test/loginForm.test.tsx:42]: Should this be `===`?

File: src/resources/js/test/loginForm.test.tsx [hunk 3]
```diff
@@ -1,27 +1,136 @@
...
+    const r: Response = new Response(strm, { status: 401, statusText: 'Unauthorized' });
+    throw new ResponseError(r, 'Response returned an error code');
+  }
+}
+
+async function mockAuthLogout(): Promise<void> {
+  savedAccessToken = null;
+}
+
+vi.spyOn(ApiClientWrapper.api, 'token', 'get').mockImplementation(mockGetToken);
```
Comment (jbraswell) [inline src/resources/js/test/loginForm.test.tsx:73]: Can move these to a `beforeAll` and add `mockReset`.

File: src/resources/js/test/loginForm.test.tsx (hunk 2 above)
Comment (alanb2718) [reply src/resources/js/test/loginForm.test.tsx:42]: Yes, that's better!

File: src/resources/js/test/loginForm.test.tsx (hunk 3 above)
Comment (alanb2718) [reply src/resources/js/test/loginForm.test.tsx:73]: OK done!

**Review by jbraswell — APPROVED** [review]

**Pull request merged**